### PR TITLE
feat: Integration tests for down rpc

### PR DESCRIPTION
**Pull request by koonopek**

### CODE REVIEW

File: packages/integration/src/framework/integration-test-utils.ts [hunk 1]
```diff
@@ -19,9 +21,7 @@ export const stopChild = (
   }
 };
 
-export const sleep = async (millis: number) => {
-  await new Promise((resolve) => setTimeout(resolve, millis));
-};
+export const sleep = (ms: number) => RedstoneCommon.sleep(ms);
```
Comment (rpiszczatowski) [inline packages/integration/src/framework/integration-test-utils.ts:24]: Maybe we can get rid of sleep altogether when it is a simple pass-through method with no value added?

File: packages/integration/src/framework/integration-test-utils.ts [hunk 2]
```diff
@@ -32,24 +32,27 @@ export const printDotenv = (label: string, filePath: string) => {
...
     }
     debug(`${filePath} is not present, waiting, tries left ${tries}...`);
     await sleep(SLEEP_TIME_MILLISECONDS);
   }
-  debug(`${filePath} bacame available`);
+  debug(`${filePath} became available`);
 };
 
 export const waitForUrl = async (url: string, tries = 5) => {
   // eslint-disable-next-line no-constant-condition,@typescript-eslint/no-unnecessary-condition
```
Comment (rpiszczatowski) [inline packages/integration/src/framework/integration-test-utils.ts:44]: obsoleted comment

File: packages/integration/src/relayer-conditions/relayer-one-rpc-down.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+      BTC: 16000,
+      __DEFAULT__: 42,
+    },
+    oracleNodeInstance
+  );
+  await startAndWaitForOracleNode(oracleNodeInstance, [cacheLayerInstance]);
+  await waitForDataAndDisplayIt(cacheLayerInstance);
+
+  await startAndWaitForHardHat(firstHardhatInstance);
+  // await startAndWaitForHardHat(secondHardhatInstance, { port: 8989 });
```
Comment (rpiszczatowski) [inline packages/integration/src/relayer-conditions/relayer-one-rpc-down.ts:51]: Should this be commented out?

**Review by rpiszczatowski — APPROVED** [review]
minor comments.

File: packages/integration/src/framework/integration-test-utils.ts (hunk 2 above)
Comment (koonopek) [reply packages/integration/src/framework/integration-test-utils.ts:44]: done

File: packages/integration/src/framework/integration-test-utils.ts (hunk 1 above)
Comment (koonopek) [reply packages/integration/src/framework/integration-test-utils.ts:24]: done

File: packages/integration/src/relayer-conditions/relayer-one-rpc-down.ts (hunk 3 above)
Comment (koonopek) [reply packages/integration/src/relayer-conditions/relayer-one-rpc-down.ts]: Good catcch :)

**Comment by koonopek** [comment]
Omg, I dont know why but hardhat doesnt work, when you use `localhost` instead of `127.0.0.1`

File: .github/workflows/monorepo-integration-tests.yml [hunk 4]
```diff
@@ -4,12 +4,16 @@ on:
   push:
     branches: ["*"]
```
Comment (rpiszczatowski) [inline .github/workflows/monorepo-integration-tests.yml:5]: Why ["*"] instead of "**". Now, these tests won't run on code review branches, I believe.

**Review by rpiszczatowski — DISMISSED** [review]

**Review by cehali — APPROVED** [review]
LGTM

File: .github/workflows/monorepo-integration-tests.yml (hunk 4 above)
Comment (koonopek) [reply .github/workflows/monorepo-integration-tests.yml]: It is stated in schema and documentation that value has to be array not string.
It was "*" so I have changed to ["*"].
But you are correct it should be ["**"], changed

**Review by rpiszczatowski — APPROVED** [review]

**Pull request merged**